cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaSetOperator.java

```diff
@@ -72,4 +88,55 @@ public static boolean needsRollingUpdate(StatefulSetDiff diff) {
                 .rollingRestart(podNeedsRestart);
     }
 
+    /**
+     * @param sts Stateful set to which kafka pod belongs
+     * @param pod Specific kafka pod
+     * @return a future which contains map with all kafka properties including their values
+     */
+    public Future<Map<ConfigResource, Config>> getCurrentConfig(StatefulSet sts, Pod pod) {
+        String cluster = sts.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL);
+        String namespace = sts.getMetadata().getNamespace();
+        Future<Secret> clusterCaKeySecretFuture = secretOperations.getAsync(
+                namespace, KafkaResources.clusterCaCertificateSecretName(cluster));
+        Future<Secret> coKeySecretFuture = secretOperations.getAsync(
+                namespace, ClusterOperator.secretName(cluster));
+        int podId = Integer.parseInt(pod.getMetadata().getName().substring(pod.getMetadata().getName().lastIndexOf("-") + 1));
+        return CompositeFuture.join(clusterCaKeySecretFuture, coKeySecretFuture).compose(compositeFuture -> {
+            Secret clusterCaKeySecret = compositeFuture.resultAt(0);
+            if (clusterCaKeySecret == null) {
+                return Future.failedFuture(missingSecretFuture(namespace, KafkaCluster.clusterCaKeySecretName(cluster)));
+            }
+            Secret coKeySecret = compositeFuture.resultAt(1);
+            if (coKeySecret == null) {
+                return Future.failedFuture(missingSecretFuture(namespace, ClusterOperator.secretName(cluster)));
+            }
+            return getCurrentConfig(podId, namespace, cluster, clusterCaKeySecret, coKeySecret);
+        });
+    }
+
+    public Future<Map<ConfigResource, Config>> getCurrentConfig(int podId, String namespace, String cluster,
+                                         Secret clusterCaCertSecret, Secret coKeySecret) {
+        Promise<Map<ConfigResource, Config>> futRes = Promise.promise();
+        String hostname = KafkaCluster.podDnsName(namespace, cluster, KafkaCluster.kafkaPodName(cluster, podId)) + ":" + KafkaCluster.REPLICATION_PORT;
+        AdminClient ac = adminClientProvider.createAdminClient(hostname, clusterCaCertSecret, coKeySecret);
+        ConfigResource resource = new ConfigResource(ConfigResource.Type.BROKER, String.valueOf(podId));
+        DescribeConfigsResult configs = ac.describeConfigs(Collections.singletonList(resource));
+        ac.close();
+        Map<ConfigResource, Config> config = null;
+        try {
+            config = configs.all().get(1000, TimeUnit.MILLISECONDS);
```

Comment:
a couple of considerations, maybe we should take a look at the `AdminClient` Kafka source code.
Is it going to work if you close the admin client instance `ac.close()` before getting the result?
Is it a blocking operation? should we need a more async way with Vert.x execute blocking code?

Reply:
I totally agree with you. Consider this piece of code to be here just for the testing purposes (you see it is not called anywhere). I used it to extract the "current" Kafka configuration. In the final phase we could use some of it but as you say... we have to polish it.